src/file_operation.rs

```diff
@@ -87,13 +88,13 @@ pub fn try_add_lines(new_todo: &str) -> io::Result<()> {
 ///     None => println!("Todo not found or error reading file"),
 ///  }
 /// ```
-pub fn find_index_opt(searched_todo: &String) -> Option<usize> {
+pub fn find_index_opt(searched_todo: &String) -> Result<Option<usize>, Error> {
     match try_read_lines("todo.txt") {
         Err(why) => {
-            eprintln!("couldn't read the lines : {}", why);
-            None
+            println!("couldn't read the lines : {}", why);
```

Comment:
Why do you print here? You can let a caller handle how to treat this error.

Reply:
You are right, it does not need to print here.